Assets/MixedRealityToolkit/Extensions/GameObjectExtensions.cs

```diff
@@ -158,5 +158,47 @@ public static void DestroyGameObject(GameObject gameObject, float t = 0.0f)
         {
             UnityObjectExtensions.DestroyObject(gameObject, t);
         }
+
+        /// <summary>
+        /// Checks if any MonoBehaviour on the given GameObject is using the RequireComponentAttribute requiring type T
+        /// </summary>
+        /// <typeparam name="T">The potentially required component</typeparam>
+        /// <param name="gameObject">the GameObject requiring the component</param>
+        /// <param name="requiringTypes">A list of types that do require the component in question</param>
+        /// <returns></returns>
+        public static bool IsComponentRequired<T>(this GameObject gameObject, List<Type> requiringTypes) where T : Component
+        {
+            var genericType = typeof(T);
+            bool requiringComponent = false;
+
+            foreach (var monoBehaviour in gameObject.GetComponents<MonoBehaviour>())
+            {
+                if (monoBehaviour == null)
```

Comment:
Out of curiosity, what are scenarios where the monoBehaviour ends up being null when getting returned? Object/monobehavior destruction?

Reply:
Just ran into this. Rarely happens, but it does when the GameObject has a missing script component on it.

Reply:
Makes sense, thanks!